language/move-stdlib/nursery/BitVector.move

```diff
@@ -0,0 +1,225 @@
+module 0x1::BitVector {
+    use 0x1::Vector;
+
+    /// The provided index is out of bounds
+    const EINDEX: u64 = 0;
+
+    struct BitVector has copy, drop, store {
+        length: u64,
+        bit_field: vector<u128>,
+    }
+
+    public fun new(length: u64): BitVector {
+        let num_words = (length + 127) / 128;
+        let bit_field = Vector::empty();
+        while (num_words > 0) {
+            Vector::push_back(&mut bit_field, 0u128);
+            num_words = num_words - 1;
+        };
+
+        BitVector {
+            length,
+            bit_field,
+        }
+    }
+
+    /// Set the bit at `bit_index` in the `bitvector` regardless of its previous state.
+    public fun set(bitvector: &mut BitVector, bit_index: u64) {
+        assert(bit_index < bitvector.length, EINDEX);
+        let (inner_index, inner) = bit_index(bitvector, bit_index);
+        *inner = *inner | 1u128 << (inner_index as u8);
+    }
+
+    /// Unset the bit at `bit_index` in the `bitvector` regardless of its previous state.
+    public fun unset(bitvector: &mut BitVector, bit_index: u64) {
+        assert(bit_index < bitvector.length, EINDEX);
+        if (index(bitvector, bit_index)) {
+            let (inner_index, inner) = bit_index(bitvector, bit_index);
+            // Having negation would be nice here...
+            *inner = *inner ^ (1u128 << (inner_index as u8))
```

Comment:
If you change this to:
`*inner ^ (*inner & (1u128 << (inner_index as u8)))`
(so that you're xor'ing with a copy of the same bit value), then you can avoid the conditional.